src/librustdoc/html/render.rs

```diff
@@ -449,7 +488,7 @@ pub fn run(mut krate: clean::Crate,
     cx.krate(krate)
 }
 
-fn build_index(krate: &clean::Crate, cache: &mut Cache) -> io::Result<String> {
+fn build_index(krate: &clean::Crate, cache: &mut Cache) -> Result<String, Error> {
```

Comment:
This shouldn't need to return a `Result` any more as no I/O errors can happen within I believe